src/components/Base/Switch/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './index.scss';
+
+export default class Switch extends PureComponent {
+  static propTypes = {
+    type: PropTypes.string,
+    onText: PropTypes.string,
+    offText: PropTypes.string,
+    onChange: PropTypes.func,
+    disabled: PropTypes.bool,
+    checked: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    type: 'primary',
+    onChange: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      on: props.checked,
```

Comment:
prop is set by external component, only readable to own component, maybe:
```
this.setState({on: false})
```

Reply:
Here is the use of props to initialize the component internal state.
Props can control the component status.

Reply:
when use props to initialize the component internal state, better to add ``componentWillReceiveProps`` to avoid multi-times render when the state changed while the props changed too.

Reply:
got it